experimental/OscarDB/src/OscarDB.jl

```diff
@@ -0,0 +1,254 @@
+  return Database(client[dev ? OSCAR_DEV_DB : OSCAR_DB])
+end
+
+"""
+    getindex(db::Database, name::AbstractString)
+
+Return a `Oscar.OscarDB.Collection` instance
+from `db` with the given `name`.
+
+# Examples
+```julia-repl
+julia> Oscar.OscarDB.get_collection_names(db)
+4-element Vector{String}:
+ "zzlattices"
+ "LeechPairs"
+ "Surfaces"
+ "TransitiveSimplicialComplexes"
+
+julia> c = db["LeechPairs"];
+
+julia> length(c)
+290
+```
+"""
+Base.getindex(db::Database, name::AbstractString) = Collection(db.mdb[name])
+
+"""
+    length(c::Collection, d::Dict=Dict())
+
+Count documents in a collection `c` matching the criteria given by `d`.
+
+# Examples
+
+Same as above, but faster.
+
+```julia-repl
+julia> db = Oscar.OscarDB.get_db();
+
+julia> tscit = Oscar.OscarDB.find(db["TransitiveSimplicialComplexes"], Dict("data.betti_numbers" => ["0", "0", "0", "1"]));
+
+julia> length(tscit)
+63
+```
+"""
+function Base.length(c::Collection, d::Dict=Dict())
+  return Base.length(c.mcol, Mongoc.BSON(d))
+end
+
+@doc raw"""
+    find(c::Collection{T}, d::Dict=Dict(); opts::Union{Nothing, Dict})
+
+Returns an iterator over documents in the collection `c` which match the criteria given by `d`.
+Apply search options `opts`.
+
+# Examples
+
+Here we show how to find all vertex-transitive 3-dimensional rational homology spheres in that database (i.e., with up to 15 vertices). 
+
+The dimension $d$ of the manifold is here implicitly given as the length of the vector of Betti numbers less one.
+
+```julia-repl
+julia> db = Oscar.OscarDB.get_db();
+
+julia> tscit = Oscar.OscarDB.find(db["TransitiveSimplicialComplexes"], Dict("data.betti_numbers" => ["0", "0", "0", "1"]));
+
+julia> length([tsc for tsc in tscit])
+63
+```
+"""
+function Mongoc.find(c::Collection, d::Dict=Dict();
+                     opts::Union{Nothing, Dict}=nothing)
+  return Cursor(Mongoc.find(c.mcol, Mongoc.BSON(d); options=(isnothing(opts) ? nothing : Mongoc.BSON(opts))))
+end
+
+@doc raw"""
+    find_one(c::Collection{T}, d::Dict=Dict(); opts::Union{Nothing, Dict})
+
+Return one document from a collection `c` matching the criteria given by `d`.
+`T` can be chosen from `Polymake.BigObject` and `Mongoc.BSON`.
+Apply search options `opts`.
+
+# Examples
+
+Here we show how to find one vertex-transitive combinatorial surface with reduced rational Betti numbers $\tilde\beta_0 = \beta_1 = \beta_2 = 0$.
+The dimension of the manifold is here implicitly given as the length of the vector of Betti numbers less one.
+
+In this case we find the unique six-vertex triangulation of the real projective plane.
+
+```jldoctest
+julia> db = Oscar.OscarDB.get_db();
+
+julia> tsc = Oscar.OscarDB.find_one(db["TransitiveSimplicialComplexes"], Dict("data.betti_numbers" => ["0", "0", "0"]));
+
+julia> facets(tsc.complex)
```

Comment:
```suggestion
julia> facets(simplicial_complex(tsc))
```
Using an accessor function is preferred.